VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,826 @@
+# lowRISC SystemVerilog Coding Style Guide for Design Verification
+
+This document is a supplemental Style Guide to the [lowRISC Verilog style
+guide](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md),
+with emphasis on writing code for Design Verification (DV) in SystemVerilog,
+following the UVM methodology.
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Naming and Style](#naming-and-style)
+* [Directory Structure and Packages](#directory-structure-and-packages)
+* [UVM Guidelines](#uvm-guidelines)
+  * [Class Definitions](#class-definitions)
+  * [UVM new functions](#uvm-new-functions)
+  * [uvm_scoreboard usage](#uvm_scoreboard-usage)
+  * [uvm_agent usage](#uvm_agent-usage)
+  * [uvm_driver usage](#uvm_driver-usage)
+  * [Macro Usage](#macro-usage)
+  * [Factory](#factory)
+  * [Configuration Mechanism](#configuration-mechanism)
+  * [Sequences](#sequences)
+  * [Sequence Items](#sequence-items)
+  * [Objections and Coordinating End of Test](#objections-and-coordinating-end-of-test)
+  * [Logging and Print Messages](#logging-and-print-messages)
+  * [DPI and C Connections](#dpi-and-c-connections)
+* [SystemVerilog Language Features](#systemverilog-language-features)
+  * [Randomization](#randomization)
```

Comment:
Add a section on UVM RAL. Want to discourage the use of uvm_reg::set() followed by uvm_reg::update() since they are non-atomic operations which can cause race conditions if there are parallel threads.

Reply:
Agree with @sriyerg. It'd better program the whole register via csr_wr instead of (set + update) combination

Reply:
This section has been added under the 'UVM Guidelines' heading, thanks for the suggestion!